bfdd/bfd.c

```diff
@@ -82,6 +80,8 @@ void gen_bfd_key(struct bfd_key *key, struct sockaddr_any *peer,
 		strlcpy(key->ifname, ifname, sizeof(key->ifname));
 	if (vrfname && vrfname[0])
 		strlcpy(key->vrfname, vrfname, sizeof(key->vrfname));
+	else
+		strlcpy(key->vrfname, VRF_DEFAULT_NAME, sizeof(key->vrfname));
 }
```

Comment:
I have an issue with that.
As VRF_DEFAULT_NAME can change ( if you perform zebra -o vrf0) for isntance, you will observe that the bfd session is yet attached to "default" vrf, and not "vrf0" vrf.
this is the same vrf pointer, but not the same key.vrfname[].
Any possibilities to empty the key->vrfname field ?

Reply:
The problem leaving this field empty is that the sessions keys (in the YANG model) will also have empty strings.

How is `ripd` or `ripngd` northbound handling this?

Reply:
I currently look at rip behaviour, and it looks like that there is an issue.
I will let you know my conclusion later.
In the meantime, I think we can go ahead with that .